lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -907,156 +820,109 @@ def _apply_params(
-            else:
-                optimised_coeffs_at_date = (
-                    optimised_coeffs[date])
-
-                # Assigning coefficients to coefficient names.
-                if len(optimised_coeffs_at_date) == len(coeff_names):
-                    optimised_coeffs_at_date = dict(
-                        zip(coeff_names, optimised_coeffs_at_date))
-                elif len(optimised_coeffs_at_date) > len(coeff_names):
-                    excess_beta = (
-                        optimised_coeffs_at_date[len(coeff_names):].tolist())
-                    optimised_coeffs_at_date = (
-                        dict(list(zip(coeff_names, optimised_coeffs_at_date))))
-                    optimised_coeffs_at_date["beta"] = np.array(
-                        [optimised_coeffs_at_date["beta"]]+excess_beta,
-                        dtype=np.float32)
-                else:
-                    msg = ("Number of coefficient names {} with names {} "
-                           "is not equal to the number of "
-                           "optimised_coeffs_at_date values {} "
-                           "with values {} or the number of "
-                           "coefficients is not greater than the "
-                           "number of coefficient names. Can not continue "
-                           "if the number of coefficient names out number "
-                           "the number of coefficients".format(
-                               len(coeff_names), coeff_names,
-                               len(optimised_coeffs_at_date),
-                               optimised_coeffs_at_date))
-                    raise ValueError(msg)
-
-                if predictor_of_mean_flag.lower() in ["mean"]:
-                    # Calculate predicted mean = a + b*X, where X is the
-                    # raw ensemble mean. In this case, b = beta.
-                    beta = [optimised_coeffs_at_date["alpha"],
-                            optimised_coeffs_at_date["beta"]]
-                    forecast_predictor_flat = (
-                        forecast_predictor_at_date.data.flatten())
-                    new_col = np.ones(forecast_predictor_flat.shape,
-                                      dtype=np.float32)
-                    all_data = np.column_stack(
-                        (new_col, forecast_predictor_flat))
-                    predicted_mean = np.dot(all_data, beta)
-                    calibrated_forecast_predictor_at_date = (
-                        forecast_predictor_at_date)
-                elif predictor_of_mean_flag.lower() in ["realizations"]:
-                    # Calculate predicted mean = a + b*X, where X is the
-                    # raw ensemble mean. In this case, b = beta^2.
-                    beta = np.concatenate(
-                        [[optimised_coeffs_at_date["alpha"]],
-                         optimised_coeffs_at_date["beta"]**2])
-                    forecast_predictor = (
```

Comment:
Why have you got rid of this line?

Reply:
This line was not having any effect as `forecast_predictor` is not used again in this function.